v2/datastream-to-spanner/terraform/samples/pre-configured-conn-profiles/terraform.tfvars

```diff
@@ -0,0 +1,88 @@
+# Common Parameters
+common_params = {
+  project                         = "<YOUR_PROJECT_ID>"
+  region                          = "<YOUR_GCP_REGION>"
+  migration_id                    = "<YOUR_MIGRATION_ID>" #Will be used as a prefix for all resources, auto-generated if not specified
+  add_policies_to_service_account = "<TRUE/FALSE>"        # This will decide if roles will be attached to service accounts or not.
+}
+
+# Datastream Parameters
+datastream_params = {
+  source_connection_profile_id  = "<YOUR_SOURCE_CONNECTION_PROFILE_ID>"
+  target_connection_profile_id  = "<YOUR_TARGET_CONNECTION_PROFILE_ID>"
+  target_gcs_bucket_name        = "<YOUR_GCS_BUCKET_NAME>"
+  pubsub_topic_name             = "live-migration" # Or your custom topic name
+  stream_id                     = "mysql-stream"   # Or your custom stream ID
+  max_concurrent_cdc_tasks      = 50               # Adjust as needed
+  max_concurrent_backfill_tasks = 50               # Adjust as needed
+  mysql_databases = [
+    {
+      database = "<YOUR_DATABASE_NAME>"
+      tables   = []
+      # Optionally list specific tables, or remove "tables" all together for all tables
+    },
+    # Add more database objects if needed
+  ]
+}
+
+# Dataflow Parameters
+dataflow_params = {
+  template_params = {
+    shadow_table_prefix = "<YOUR_SHADOW_TABLE_PREFIX>"
+    # e.g., "shadow_"
+    create_shadow_tables = true # true or false
+    rfc_start_date_time  = "<YYYY-MM-DDTHH:MM:SSZ>"
+    # e.g., "2023-12-31T12:00:00Z" (optional)
+    file_read_concurrency = 10 # Adjust as needed
+    session_file_path     = "<YOUR_SESSION_FILE_PATH>"
+    # Path to your session file (optional)
+    spanner_instance_id = "<YOUR_SPANNER_INSTANCE_ID>"
+    spanner_database_id = "<YOUR_SPANNER_DATABASE_ID>"
+    spanner_host        = "https://<YOUR_REGION>-spanner.googleapis.com"
+    # Replace <YOUR_REGION>
+    dead_letter_queue_directory = "<YOUR_DLQ_DIRECTORY>"
+    # e.g., "gs://<YOUR_BUCKET>/dlq" (optional)
+    dlq_retry_minutes   = 10 # Adjust as needed
+    dlq_max_retry_count = 3  # Adjust as needed
+    datastream_root_url = "<YOUR_DATASTREAM_ROOT_URL>"
+    # Base URL of your Datastream API (optional)
+    datastream_source_type           = "MYSQL"
+    round_json_decimals              = false
+    run_mode                         = "STREAMING"
+    transformation_context_file_path = "<YOUR_TRANSFORMATION_FILE_PATH>"
```

Comment:
how do folks know that this is optional ? Can we comment in the tf file to make it clearer without having to look into documentation ?

Reply:
This can be determined from `variables.tf` which declares if each variable is optional or mandatory, along with defaults.

There is a `terraform_simple.tfvars` as well which demonstrates the minimal configuration the user needs to do.